src/app/shared/components/user-profile-header/user-profile-header.component.ts

```diff
@@ -1,38 +1,43 @@
 import { Component, OnInit } from '@angular/core';
 import { Router, RouterLink, RouterLinkActive } from '@angular/router';
 import { ProfileService } from '../../../features/profile-management/services/profile.service';
-import { Subscription } from 'rxjs';
+import { Observable, Subscription } from 'rxjs';
 import { User } from '../../../features/profile-management/models/user';
+import { IProfile } from '../../../features/profile-management/models/profile';
+import { CommonModule } from '@angular/common';
 
 @Component({
   selector: 'app-user-profile-header',
   standalone: true,
-  imports: [RouterLink, RouterLinkActive],
+  imports: [RouterLink, RouterLinkActive, CommonModule],
   templateUrl: './user-profile-header.component.html',
   styleUrl: './user-profile-header.component.sass'
 })
 export class UserProfileHeaderComponent implements OnInit{
   loggedInUser: User | null = null
-  private subscription: Subscription = new Subscription()
   loggedInSubcription!: Subscription
+  userProfile!: Observable<IProfile>
+  defaultImage = 'assets/images/default-avatar.png'
 
   constructor(private profileService: ProfileService, private router: Router) {}
 
   ngOnInit(): void {
     this.loggedInSubcription = this.profileService.loggedInUser$.subscribe((user) => {
       this.loggedInUser = user
     })
+
+    this.userProfile = this.profileService.getClient()
   }
   
   navigateToProfile(): void {
     const [role] = this.loggedInUser?.role || []
     if (role === 'ROLE_ADMIN') {
       this.router.navigate(['/admin/dashboard/profile'])
     }
-    else if (role === 'SEEKER') {
+    else if (role === 'ROLE_SEEKER') {
       this.router.navigate(['/profile'])
     }
-    else {
+    else if (role === 'ROLE_PROVIDER') {
       this.router.navigate(['/provider/dashboard'])
     }
   }
```

Comment:
you can create a constant for your roles, that will help you make change in a specific file when you run into and issue or there is a typo

something like what we have below
export const ROLE_PROVIDER = 'ROLE_PROVIDER'